app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/editRecipe/ChangeCommand.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * KitchenPlanerApp is the android app frontend for the KitchenPlaner, a tool
+ * to cooperatively plan a meal plan for a campout.
+ * Copyright (C) 2023-2024 Arne Kuchenbecker, Antonia Heiming, Anton Kadelbach, Sandra Lanz
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+package com.scouts.kitchenplaner.ui.viewmodel.editRecipe
+
+import com.scouts.kitchenplaner.model.entities.Recipe
+import com.scouts.kitchenplaner.model.usecases.EditRecipe
+import com.scouts.kitchenplaner.ui.state.EditRecipeState
+
+/**
+ * The command to make changes on a recipe. It is used in a command pattern.
+ *
+ * @param recipe The recipe to which the changes belong
+ */
+abstract class ChangeCommand(protected open val recipe: Recipe) {
```

Comment:
I would suggest to move ```EditRecipe``` and ```EditRecipeState``` to the constructor and create commands via a factory. The factory could then take the recipe and ```EditRecipe``` as constructor parameters and create its own ```EditRecipeState``` in its constructor. You would need to create a new factory each time the user switches into edit mode (where you currently create the new state). The factory then inserts the three parameters that are identical when creating the commands.